pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -258,44 +259,54 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 	return nil
 }
 
-func finalizeIfNeeded(ctx context.Context, stream *imagev1.ImageStream, srcClusterName string, clients map[string]ctrlruntimeclient.Client) error {
-	finalizerSet := sets.NewString(stream.Finalizers...)
-	if !finalizerSet.Has(finalizerName) {
-		return nil
-	}
-	for clusterName, client := range clients {
-		if clusterName == srcClusterName {
-			continue
+func finalizeIfNeeded(ctx context.Context, sourceImageStream *imagev1.ImageStream, clients map[string]ctrlruntimeclient.Client, log *logrus.Entry) (bool, error) {
+	foundDeleteTimestamp := false
+	for _, client := range clients {
+		stream := &imagev1.ImageStream{}
+		if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Name: sourceImageStream.Name, Namespace: sourceImageStream.Namespace}, stream); err != nil {
+			if apierrors.IsNotFound(err) {
+				continue
+			}
+			return false, err
+		}
+		if stream.DeletionTimestamp != nil {
+			foundDeleteTimestamp = true
+			break
 		}
+	}
+	if !foundDeleteTimestamp {
+		return false, nil
+	}
 
-		isToDelete := &imagev1.ImageStream{}
-		if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Name: stream.Name, Namespace: stream.Namespace}, isToDelete); err != nil {
+	for _, client := range clients {
+		stream := &imagev1.ImageStream{}
+		if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Name: sourceImageStream.Name, Namespace: sourceImageStream.Namespace}, stream); err != nil {
 			if apierrors.IsNotFound(err) {
 				continue
 			}
-			return err
+			return false, err
 		}
-
-		if err := ensureRemoveFinalizer(ctx, isToDelete, client); err != nil {
-			return err
+		if err := ensureRemoveFinalizer(ctx, stream, client, log); err != nil {
+			return false, err
 		}
-
-		// populate deleting to all clusters except srcClusterName
-		if err := client.Delete(ctx, isToDelete); err != nil && !apierrors.IsNotFound(err) {
-			return err
+		if stream.DeletionTimestamp == nil {
+			log.Debug("deleting imagestream")
+			// populate deleting to all clusters except srcClusterName
```

Comment:
That comment isn't correct anymore I think?

Reply:
Updated.